unlock-app/src/services/walletService.js

```diff
@@ -91,23 +95,58 @@ export default class WalletService extends EventEmitter {
   }
 
   /**
-   * Function which yields the address of the account on the provider or creates a key pair.
+   * Poll to see if account has changed
    */
-  getAccount() {
-    const getOrCreateAccount = this.web3.eth.getAccounts().then(accounts => {
-      if (accounts.length === 0) {
-        return this._createAccount()
-      } else {
-        return Promise.resolve(accounts[0]) // defaults to the first account for now
-      }
-    })
+  async pollForAccountChange(isServer) {
+    if (isServer) return
+    try {
+      await delayPromise(POLLING_INTERVAL)
+    } catch (e) {
+      return // if the timeout throws, we don't start the loop, this is primarily for testing
+    }
+    try {
+      this.account = await this.checkForAccountChange(this.account)
```

Comment:
Same here, is there any reason it would throw?

Reply:
yes, if web3 throws.

Reply:
(which it does inside an iframe in coinbase wallet, getAccount() freaks out)

Reply:
Is there a way to only catch that error? I m not a huge fan of blanket `catch (e)` which catches and swallows everything.

Reply:
I'm pretty sure it's a generic Error, so I don't think so. Also, there may be new errors introduced later. The solution is a TODO that says "log error here" for when we implement Akeem's idea of logging